src/weapons.py

```diff
@@ -18,10 +22,52 @@ class DamageType(Enum):
     thunder = "thunder"
 
 
+@dataclass
+class RollAffects:
+    immunities: List[DamageType]
+    resistances: List[DamageType]
+    vulnerabilities: List[DamageType]
+
+
+class CompoundRollResult:
```

Comment:
Is this class specific to damage rolls?
if so, maybe "CompoundDamageRollResult" is more suitable.
Also, have you examined its relationship to the dice module? Maybe it can inherit "RollResult"

Reply:
I changed the name to `CompoundDamageRollResult`.
I don't think it should inherit from RollResult, as they don't share the same interface. We could create a parent interface class (abstract class) and unify the interface.